turbopack/crates/turbopack-resolve/src/ecmascript.rs

```diff
@@ -45,20 +45,16 @@ pub fn apply_esm_specific_options(
     options: Vc<ResolveOptions>,
     reference_type: Value<ReferenceType>,
 ) -> Vc<ResolveOptions> {
-    apply_esm_specific_options_internal(
-        options,
-        matches!(
-            reference_type.into_value(),
-            ReferenceType::EcmaScriptModules(EcmaScriptModulesReferenceSubType::ImportWithType(_))
-        ),
-    )
+    apply_esm_specific_options_internal(options, reference_type)
 }
 
 #[turbo_tasks::function]
 async fn apply_esm_specific_options_internal(
     options: Vc<ResolveOptions>,
-    clear_extensions: bool,
+    reference_type: Value<ReferenceType>,
 ) -> Result<Vc<ResolveOptions>> {
```

Comment:
This function was specifically added so that the `ResolveOptions` don't depend on the `ReferenceType`. Now, you will get hundrets or thousands or unique `ResolveOptions` and much less cache hits on the actual `resolve` turbotask. So if possible, we should retain the old behavior somehow.

Reply:
Will the new code work?

Reply:
It will still resolve many things again and again if the `Export(str)` is different.
I'm not entirely convinced that the ExportUsage really has to be on the resolve options, but I'll look at all of your changes tomorrow, I don't completely understand yet where you read this information.